File: app/src/main/java/com/yeonsik/fitnessapp/development/BodyProfile.java
package com.yeonsik.fitnessapp.development;

public final class BodyProfile {
    public final Integer heightCm;
    public final String createdAt;
    public final String updatedAt;

    public BodyProfile(Integer heightCm, String createdAt, String updatedAt) {
        this.heightCm = requireHeight(heightCm);
        this.createdAt = normalizeTimestamp(createdAt);
        this.updatedAt = normalizeTimestamp(updatedAt);
    }

    public static BodyProfile empty() {
        return new BodyProfile(null, "", "");
    }

    public boolean isConfigured() {
        return heightCm != null;
    }

    public String heightLabelKo() {
        return isConfigured() ? heightCm + "cm" : "미설정";
    }

    private static Integer requireHeight(Integer value) {
        if (value == null) {
            return null;
        }
        if (value < 50 || value > 300) {
            throw new IllegalArgumentException("키는 50~300cm 범위여야 합니다.");
        }
        return value;
    }

    private static String normalizeTimestamp(String value) {
        return value == null ? "" : value.trim();
    }
}

File: shared/src/commonMain/kotlin/com/yeonsik/fitness/shared/feature/body/model/BodyProfile.kt
package com.yeonsik.fitness.shared.feature.body.model

import kotlin.jvm.JvmField
import kotlin.jvm.JvmStatic

/** Immutable body profile read model. Storage and write ownership remain platform-side. */
class BodyProfile(
    @JvmField val heightCm: Int?,
    createdAt: String?,
    updatedAt: String?
) {
    init {
        require(heightCm == null || heightCm in 50..300) {
            "키는 50~300cm 범위여야 합니다."
        }
    }

    @JvmField val createdAt: String = normalizeTimestamp(createdAt)
    @JvmField val updatedAt: String = normalizeTimestamp(updatedAt)

    fun isConfigured(): Boolean = heightCm != null

    fun heightLabelKo(): String = heightCm?.let { "${it}cm" } ?: "미설정"

    companion object {
        @JvmStatic
        fun empty(): BodyProfile = BodyProfile(null, "", "")

        private fun normalizeTimestamp(value: String?): String = value?.trim() ?: ""
    }
}
